kolibri/plugins/management/assets/src/vue/user-page/user-edit-modal.vue

```diff
@@ -20,20 +20,60 @@
         <div class="user-field">
           <label for="user-role"><span class="visuallyhidden">User Role</span></label>
           <select v-model="role_new" id="user-role">
-            <option value="learner" selected> Learner </option>
-            <option value="admin"> Admin </option>
+            <option :selected="role_new == admin ? true : false" v-if="role_new" value="learner"> Learner </option>
+            <option :selected="role_new == admin ? true : false" value="admin"> Admin </option>
           </select>
         </div>
 
+        <div class="advanced-options" v-if="!pw_reset && !usr_delete">
+          <button @click="pw_reset=!pw_reset"> Reset Password </button>
+          <button @click="usr_delete=!usr_delete"> Delete User</button>
+        </div>
+
+        <hr class="end-modal">
+
+      </div>
+
+      <div @keyup.enter="editUser" v-if="pw_reset" slot="body">
+        <p>Username: <b>{{username_new}}</b></p>
         <div class="user-field">
-          <label for="password">Password</label>:
-          <input type="password" class="edit-form" id="password" required v-model="password_new" placeholder="Please type in your password.">
+          <label for="password">Enter new password</label>:
+          <input type="password" class="edit-form" id="password" required v-model="password_new">
         </div>
 
+        <div class="user-field">
+          <label for="password-confirm">Confirm new password</label>:
+          <input type="password" class="edit-form" id="password-confirm" required v-model="password_new_confirm">
+        </div>
+      </div>
+
+      <div @keyup.enter="editUser" v-if="usr_delete" slot="body">
+        <div class="user-field">
+          <p> Are you sure you want to delete
+          <b>{{username_new}}</b>?
+          </p>
+        </div>
       </div>
 
       <div slot="footer">
-        <button class="confirm-btn" type="button" @click="editUser">Confirm</button>
```

Comment:
this should probably have used three independent methods (for edit, delete, and password reset) rather than repurposing the same method and buttons